### PR TITLE
Prepare for upcoming breaking changes in FastBoot 1.0

### PR DESCRIPTION
**Pull request by simonihmig**
A few things will be gone with the upcoming FastBoot 1.0 release:
* no more `process.env.EMBER_CLI_FASTBOOT` (as there is just one unified build step)
* no `initializer/(browser|fastboot)/*` (again, just one build)

The changes here should continue to work with pre 1.0 and also support post 1.0 versions.

### CODE REVIEW

**Comment by simonihmig** [comment]
Btw, here is some context: https://github.com/ember-fastboot/ember-cli-fastboot/issues/360

Not sure what's happening with Travis again, cannot install phantomjs?

File: index.js [hunk 1]
```diff
@@ -46,8 +43,9 @@ module.exports = {
   },
 
   config(env, baseConfig) {
-    if (!env)
+    if (!env) {
```
Comment (kratiahuja) [inline index.js]: When is `env` not going to be defined?

Comment (simonihmig) [inline index.js]: Not sure tbh

**Comment by simonihmig** [comment]
The failing tests are due to https://github.com/yarnpkg/yarn/issues/3138 😞

**Comment by andreasschacht** [comment]
@simonihmig should we force travis to use a later node version?

**Comment by simonihmig** [comment]
Yes, we could so, as a workaround, but should revert once the yarn bug is fixed!

File: index.js (hunk 1 above)
Comment (andreasschacht) [inline index.js]: I can't really remember, why I introduce this guard. Maybe to eliminate any possibility of being the cause of a bug in early days. I will remove this in the next release.

**Pull request merged**